library/std/src/sys/anonymous_pipe/unix.rs

```diff
@@ -0,0 +1,111 @@
+use super::*;
+
+use crate::{
+    fs::File,
+    os::{
+        fd::{AsFd, AsRawFd, BorrowedFd, FromRawFd, IntoRawFd, OwnedFd, RawFd},
+        unix::fs::FileTypeExt,
+    },
+    sys::{
+        fd::FileDesc,
+        pipe::{anon_pipe, AnonPipe},
+    },
+    sys_common::{FromInner, IntoInner},
+};
+
+#[inline]
+pub(super) fn pipe() -> io::Result<(PipeReader, PipeWriter)> {
+    anon_pipe().map(|(rx, tx)| (PipeReader(rx), PipeWriter(tx)))
+}
+
+macro_rules! impl_traits {
+    ($name:ty) => {
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl AsFd for $name {
+            fn as_fd(&self) -> BorrowedFd<'_> {
+                self.0.as_fd()
+            }
+        }
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl AsRawFd for $name {
+            fn as_raw_fd(&self) -> RawFd {
+                self.0.as_raw_fd()
+            }
+        }
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl From<$name> for OwnedFd {
+            fn from(pipe: $name) -> Self {
+                FileDesc::into_inner(AnonPipe::into_inner(pipe.0))
+            }
+        }
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl FromRawFd for $name {
+            unsafe fn from_raw_fd(raw_fd: RawFd) -> Self {
+                Self(AnonPipe::from_raw_fd(raw_fd))
+            }
+        }
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl IntoRawFd for $name {
+            fn into_raw_fd(self) -> RawFd {
+                self.0.into_raw_fd()
+            }
+        }
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl From<$name> for Stdio {
+            fn from(pipe: $name) -> Self {
+                Self::from(OwnedFd::from(pipe))
+            }
+        }
+    };
+}
+impl_traits!(PipeReader);
+impl_traits!(PipeWriter);
+
+fn convert_to_pipe(owned_fd: OwnedFd) -> io::Result<AnonPipe> {
+    let file = File::from(owned_fd);
+    if file.metadata()?.file_type().is_fifo() {
+        Ok(AnonPipe::from_inner(FileDesc::from_inner(OwnedFd::from(file))))
+    } else {
+        Err(io::Error::new(io::ErrorKind::InvalidInput, "Not a pipe"))
+    }
+}
+
+#[unstable(feature = "anonymous_pipe", issue = "127154")]
+impl TryFrom<OwnedFd> for PipeReader {
```

Comment:
`File` implements `From<OwnedFd>` directly without any check. Should we do the same here for consistency ? 

Same question for `PipeWriter` and for `From<OwnedHandle>` on Windows.

Reply:
Hmmm, yeah perhaps it makes sense to remove it.

It does not pose any memory safety issue whatever, I can remove it

Reply:
I've removed it.